### ISSUE TITLE
main script error 

### ISSUE THREAD

**Issue by rfour92** [iquasere/KEGGCharter#6]
Hello, 

I am getting an error even when I apply this code : keggcharter.py -f test.xlsx -tcol occ -gcol gene_name -it -keggc ko -o KEGG_Test
=====
Error message  I am getting: 

Created KEGG_Test
2022-11-13 10:47:35: Arguments valid.
2022-11-13 10:47:39: Data successfully read.
Converting 2376 KEGG IDs to KOs through the KEGG API: 100%|██████████| 16/16 [00:32<00:00,  2.02s/it]
Converting 0 KOs to EC numbers through the KEGG API: 0it [00:00, ?it/s]
2022-11-13 10:48:11: Results saved to KEGG_Test/KEGGCharter_results.tsv
Getting information for 2 taxa:  50%|█████     | 1/2 [00:03<00:03,  4.00s/it]
Traceback (most recent call last):
  File "/ibex/scratch/alamourt/conda_keggcharter_env/bin/keggcharter.py", line 480, in <module>
    main()
  File "/ibex/scratch/alamourt/conda_keggcharter_env/bin/keggcharter.py", line 450, in main
    taxon_to_mmap_to_orthologs = download_resources(data, args.resources_directory, args.taxa_column, metabolic_maps)
  File "bin/keggcharter.py", line 331, in download_resources
    kegg_prefix = taxon2prefix(taxon, taxa_df)
  File "bin/keggcharter.py", line 296, in taxon2prefix
    if taxon_name.split(' (')[0] in organism_df.index:  # Homo sapiens (human) -> Homo sapiens
AttributeError: 'bool' object has no attribute 'split'

I performed the example file given along with the commands given and it worked perfectly. however in this case it doesn't seem to work, can you please advise?

**Comment by iquasere** [comment]
Hi there!

I see two issues here, one being reported and the other silently failing. "taxon_name" of your data seems to be a column of booleans, but it should be a column of species names. Also, it seems KEGGCharter could not find KOs from your KEGG IDs.

Could you please share the command you are using to run KEGGCharter, and the input file (or a subset of it)?

**Comment by rfour92** [comment]
Hello, 

thank you for your prompt reply, I highly appreciate it 

so my main goal is to be able to visualize KEGG codes of a given genera and determine the number of times each KO occurs. 

I understand this is not the main goal of this tool but I think it can work if I introduced the data as RNA transcript count. 

an example, 
#######
gene_ name | ko | occ 
gene_1 | K01923 | 43
#######
gene_name: gene name - place holder
ko = kegg codes, 
occurances: the number of times the KO was detected. 



the way I get the data is using eggnog-mapper then merge all annotation output and count the number of times each gene was detected. 

I am more than happy to supply the full file if needed.

**Comment by iquasere** [comment]
Supplying the full file AND the command used will allow me to understand what is going wrong. KEGGCharter is still very limited in what KEGG IDs it can handle, and it seems it can't handle yours. Also, seems to be some pickle on your taxonomy information. Therefore, to reproduce the error and diagnose it, I need to reproduce what you have done. 

If KEGGCharter runs fine with the example data, it is failing to interpret your particular data, and I require it to understand what is going on.

**Comment by rfour92** [comment]
[test.xlsx](https://github.com/iquasere/KEGGCharter/files/10002888/test.xlsx)


of course, 
please find the attached file and here is the command I used 
keggcharter.py -f test.xlsx -tcol occ -gcol gene_name -it -keggc ko -o KEGG_Test

thank you

**Comment by iquasere** [comment]
What is the purpose of the ```occ``` column in this data?

**Comment by rfour92** [comment]
occ stands for occurrence and it should reflect the number of times a given ko is detected, like in a pan-genome 

example: out of 50 species within genera, how many of them had glucokinase. lets assume the answer is 40. 
so for glucokinase within genus x, the occurrence is 40

**Comment by iquasere** [comment]
A couple of things here, first you should update KEGGCharter to version 0.4.1 (the latest) as with your example I found some problems I hadn't yet tackled. This version will be available soon, so please wait a bit until it is released (likely today). About your command, it should be set as 
```
keggcharter.py -f test.xlsx -it "My taxonomy" -gcol occ -tcol occ -koc ko
```
* ```-it``` parameter takes as input a value, a string, which will be used to represent the taxonomy in the maps. 
* since you are inputting KOs, you should input them with the ```-koc``` parameter, instead of the ```-keggc``` one
* the columns parameters in KEGGCharter do have weird names, but the same value should be inputted to ```-gcol``` and ```-tcol``` parameters here because those parameters concern quantification, and the only quantification here is in the "occ" column

**Comment by iquasere** [comment]
Closing this as fixed by the two commits today, if version 0.4.1 doesn't meet your needs please do inform here.
